contracts/v2/upgrader/V2Upgrader.sol

```diff
@@ -0,0 +1,170 @@
+/**
+ * SPDX-License-Identifier: MIT
+ *
+ * Copyright (c) 2018-2020 CENTRE SECZ
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+pragma solidity 0.6.11;
+
+import { SafeMath } from "@openzeppelin/contracts/math/SafeMath.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { Ownable } from "../../v1/Ownable.sol";
+import { FiatTokenV2 } from "../FiatTokenV2.sol";
+import { FiatTokenProxy } from "../../v1/FiatTokenProxy.sol";
+import { V2UpgraderHelper } from "./V2UpgraderHelper.sol";
+
+/**
+ * @title V2 Upgrader
+ * @notice Performs USDC v2 upgrade, and runs a basic sanity test in a single
+ * atomic transaction, rolling back if any issues are found. This may be
+ * overkill, but the peace of mind is worth the gas spent. By performing the
+ * upgrade atomically, it ensures that there is no disruption of service if the
+ * upgrade is not successful for some unforeseen circumstances.
+ * @dev Read docs/v2_upgrade.md
+ */
+contract V2Upgrader is Ownable {
+    using SafeMath for uint256;
+
+    FiatTokenProxy private _proxy;
+    FiatTokenV2 private _implementation;
+    V2UpgraderHelper private _helper;
```

Comment:
Should we make these public (or add getters) so they are easier to inspect/validate after we deploy?

Reply:
Sounds good.